collectorist/api/src/lib.rs

```diff
@@ -0,0 +1,83 @@
+use std::net::SocketAddr;
+use std::process::ExitCode;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::SystemTime;
+
+use futures::future::join;
+use guac::collectsub::{CollectSubClient, Entry, Filter};
+use tokio::time::interval;
+
+use trustification_infrastructure::{Infrastructure, InfrastructureConfig};
+
+mod request;
+mod server;
+
+#[derive(clap::Args, Debug)]
+#[command(about = "Run the api server", args_conflicts_with_subcommands = true)]
+pub struct Run {
+    #[arg(short, long, default_value = "0.0.0.0")]
+    pub bind: String,
+
+    #[arg(short = 'p', long = "port", default_value_t = 9919)]
+    pub port: u16,
+
+    #[command(flatten)]
+    pub infra: InfrastructureConfig,
+
+    #[arg(short = 'u', long = "csub-url", default_value = "http://localhost:2782/")]
+    pub(crate) csub_url: String,
+}
+
+impl Run {
+    pub async fn run(self) -> anyhow::Result<ExitCode> {
+        let infra = Infrastructure::from(self.infra).run("collectorist-api", |_metrics| async move {
+            let state = Self::configure()?;
+            let addr = SocketAddr::from_str(&format!("{}:{}", self.bind, self.port))?;
+
+            server::run(state, addr).await
+        });
```

Comment:
I would expect this to be the entirety of this function. You're running an HTTPServer that doesn't seem to have any handlers associated with it. It looks like the guts of your operation are in the listener loop below this code. I would expect that loop to be inside `server::run` *instead* of the HTTPServer you currently have there.

Am I misunderstanding the logic?